src/conv/problem_description.cpp

```diff
@@ -188,6 +194,36 @@ bool ProblemDescription::IsLayoutDefault() const
     }
 }
 
+bool ProblemDescription::IsLayoutNHWC() const
+{
+    if(GetSpatialDims() == 2)
+    {
+        return (in_layout == "NHWC") && (out_layout == "NHWC") && (weights_layout == "NHWC");
+    }
+    else
+    {
+        return (in_layout == "NDHWC") && (out_layout == "NDHWC") && (weights_layout == "NDHWC");
+    }
+}
+
+bool ProblemDescription::IsLayoutNCHWc() const
+{
+    return ((GetSpatialDims() == 2) && (in_layout == "NCHWc") && (out_layout == "NCHWc") &&
+            (weights_layout == "NCHWc")) ||
+           ((GetSpatialDims() == 2) && (in_layout == "NCHWc") && (out_layout == "NCHWc") &&
+            (weights_layout == "CHWNc"));
```

Comment:
[R][Quality]
```suggestion
    return IsNCHWc_NCHWc() || IsNCHWc_CHWNc;
```

Reply:
@atamazov I have moved this function from `src/problem_description.cpp` as is.
Don't we need to check `GetSpatialDims()` here?

Reply:
I do not know ;/

Reply:
@atamazov I've changed it as you proposed and kept this check